apps/web/src/components/layout/AppLayout.tsx

```diff
@@ -8,52 +8,61 @@ import { colors } from '../../design-system';
+import { SpotLight } from '../utils/Spotlight';
+import { SpotLightProvider } from '../../store/spotlightContext';
 
 export function AppLayout() {
   return (
-    <ThemeProvider>
-      <SupportChatProvider>
-        <AppShell
-          padding="lg"
-          navbar={<SideNav />}
-          header={<HeaderNav />}
-          styles={(theme) => ({
-            root: { minHeight: '100vh', position: 'relative', zIndex: 1 },
-            body: {
-              minHeight: `calc(100vh - ${HEADER_HEIGHT})`,
-              '@media (max-width: 768px)': {
-                flexDirection: 'column',
-                height: 'auto',
+    <SpotLightProvider>
+      <ThemeProvider>
+        <SupportChatProvider>
+          <AppShell
+            padding="lg"
+            navbar={<SideNav />}
+            header={<HeaderNav />}
+            styles={(theme) => ({
+              root: { minHeight: '100vh', position: 'relative', zIndex: 1 },
+              body: {
+                minHeight: `calc(100vh - ${HEADER_HEIGHT})`,
+                '@media (max-width: 768px)': {
+                  flexDirection: 'column',
+                  height: 'auto',
+                },
               },
-            },
-            main: {
-              backgroundColor: theme.colorScheme === 'dark' ? colors.BGDark : colors.BGLight,
-              minHeight: 'auto',
-              padding: '30px',
-            },
-          })}
-        >
-          <Sentry.ErrorBoundary
-            fallback={({ error, resetError, eventId }) => (
-              <>
-                Sorry, but something went wrong. <br />
-                Our team been notified about it and we will look at it asap.
-                <br />
-                <code>
-                  <small style={{ color: 'lightGrey' }}>
-                    Event Id: {eventId}.
-                    <br />
-                    {error.toString()}
-                  </small>
-                </code>
-              </>
-            )}
+              main: {
+                backgroundColor: theme.colorScheme === 'dark' ? colors.BGDark : colors.BGLight,
+                minHeight: 'auto',
+                padding: '30px',
+              },
+            })}
           >
-            <Outlet />
-          </Sentry.ErrorBoundary>
-        </AppShell>
-      </SupportChatProvider>
-    </ThemeProvider>
+            <Sentry.ErrorBoundary
+              fallback={({ error, resetError, eventId }) => (
+                <>
+                  Sorry, but something went wrong. <br />
+                  Our team been notified about it and we will look at it asap.
+                  <br />
+                  <code>
+                    <small style={{ color: 'lightGrey' }}>
+                      Event Id: {eventId}.
+                      <br />
+                      {error.toString()}
+                    </small>
+                  </code>
+                </>
+              )}
+            >
+              <RequiredAuth>
```

Comment:
I feel like this `RequiredAuth` should wrap all other components here... we should do a auth check logic as early as possible... otherwise other components, like header, and side nav can start running it's logic which in case of not logged in user should not happen